plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -837,34 +842,54 @@ private String toRemoteSchemaName(String schemaName)
         if (!caseInsensitiveNameMatching) {
             return schemaName;
         }
-        for (String remoteSchemaName : client.listDatabaseNames()) {
+        for (String remoteSchemaName : listDatabaseNames()) {
             if (schemaName.equals(remoteSchemaName.toLowerCase(ENGLISH))) {
                 return remoteSchemaName;
             }
         }
         return schemaName;
     }
 
+    private Iterable<String> listDatabaseNames()
+    {
+        return client.listDatabases()
+                .nameOnly(true)
+                .authorizedDatabasesOnly(true)
+                .map(result -> result.getString("name"));
+    }
+
     private String toRemoteTableName(String schemaName, String tableName)
     {
         verify(tableName.equals(tableName.toLowerCase(ENGLISH)), "tableName not in lower-case: %s", tableName);
         if (!caseInsensitiveNameMatching) {
             return tableName;
         }
-        for (String remoteTableName : client.getDatabase(schemaName).listCollectionNames()) {
+        for (String remoteTableName : listCollectionNames(schemaName)) {
             if (tableName.equals(remoteTableName.toLowerCase(ENGLISH))) {
                 return remoteTableName;
             }
         }
         return tableName;
     }
 
+    private List<String> listCollectionNames(String databaseName)
+    {
+        MongoDatabase database = client.getDatabase(databaseName);
+        Document cursor = database.runCommand(new Document(AUTHORIZED_LIST_COLLECTIONS_COMMAND)).get("cursor", Document.class);
+
+        List<Document> firstBatch = cursor.get("firstBatch", List.class);
```

Comment:
Would there be another batch ?

Reply:
Or a next batch ?

Reply:
I checked in a MongoDB Java driver source and by just creating 10,000 collections in MongoDB and by default MongoDB returns everything in a single response, so this implementation should be sufficient.

Reply:
Does MongoDB takes too long to create 10,000 collections ? Can we add it as a test to ensure that if the newer version returns in some batch we could handle the same ?

Reply:
Listing 10,000 collections took for me 1.14 second. Still agree a test for this case is a good idea, I will add it